_posts/2023-02-13-changelog-188.markdown

```diff
@@ -0,0 +1,86 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #188"
+date: 2023-02-13 13:00:00 +0300
+---
+
+
+## New Features
+
+* Detect new compiler errors (by [@kuksag]):
+  * [#10066] `main` function is not allowed to have generic parameters ([E0131](https://doc.rust-lang.org/error-index.html#E0131))
+  * [#10066] An inherent implementation was marked `unsafe` ([E0197](https://doc.rust-lang.org/error-index.html#E0197))
+  * [#10066] Type parameter has more than one relaxed default bound ([E0203](https://doc.rust-lang.org/error-index.html#E0203))
+  * [#9894] A pattern was declared as an argument in a foreign function declaration ([E0130](https://doc.rust-lang.org/error-index.html#E0130))
+  * [#9894] Trait methods currently cannot take patterns as arguments ([E0642](https://doc.rust-lang.org/error-index.html#E0642))
+
+* [#10057] Parse and annotate [`const_closures`](https://github.com/rust-lang/rust/issues/106003) syntax
+
+* [#8826] Support new `where` clause location for associated types
+
+* [#9987] Annotate missing lifetimes ([E0106](https://doc.rust-lang.org/error-index.html#E0106)) in function signatures
+
+* [#7489] Add debugger option to not step into stdlib sources (by [@Kobzol])
+
+  {% include gif-img.html path="/assets/posts/changelog-188/do_not_step_into_stdlib" w="600px" %}
+
+* [#9962] Show the contents of raw slice pointers (`*const [T]`, `*mut [T]`) in the debugger
+
+  <img src="/assets/posts/changelog-188/contents_of_slice_ptr.png" width="600px"/>
+
+* [#9806] Update icons for the [New UI](https://www.jetbrains.com/help/idea/new-ui.html)
+
+  <img src="/assets/posts/changelog-188/icons_new_ui.png" width="600px"/>
+
+* [#10056] Add a dedicated icon for procedural macros
+
+  <img src="/assets/posts/changelog-188/icon_proc_macro.png" width="600px"/>
+
+* [#9123] Allow excluding specific paths from auto-import via `Preferences | Editor | General | Auto Import | Rust` settings
```

Comment:
I'm afraid this might be too long for the changelog. WDYT about adding a short summary like this?
```
* [#9123] Allow excluding specific paths from auto-import via `Preferences | Editor | General | Auto Import | Rust` settings.
By default, we exclude `Borrow` and `BorrowMut` from the stdlib, as auto-import is rarely desired for them.
Tip: exclude functions or traits re-exported in irrelevant crates to apply only the original.
Read the pull request description for more information.
```

Reply:
> By default, `Borrow` and `BorrowMut` traits from stdlib are excluded, as auto-import is rarely desired for them.

This is not true since `Borrow` and `BorrowMut` are still can be auto-imported. Exclusion is only applies for their method completion

Reply:
Changed to
```
[`Borrow`](https://doc.rust-lang.org/std/borrow/trait.Borrow.html) and [`BorrowMut`](https://doc.rust-lang.org/std/borrow/trait.BorrowMut.html) methods
```